### PR TITLE
test: Add fault injection in cache fuzzer

### PR DESCRIPTION
**Pull request by zacw7**
Differential Revision: D67662693

### CODE REVIEW

File: velox/exec/fuzzer/CacheFuzzer.cpp [hunk 1]
```diff
@@ -211,6 +225,25 @@ void CacheFuzzer::initSourceDataFiles() {
       fileSizes_.emplace_back(fileSize);
     }
   }
+
+  if (FLAGS_enable_file_faulty_injection) {
+    faultyFileSystem()->setFileInjectionHook([&](FaultFileOperation* op) {
+      if (std::regex_match(op->path, kDataFileNamePattern)) {
+        // Skip errors on input data files.
```
Comment (xiaoxmeng) [inline velox/exec/fuzzer/CacheFuzzer.cpp:232]: What's the input data files here?

I think you could capture the input file directory here and match as prefix? Thanks!

File: velox/exec/fuzzer/CacheFuzzer.cpp [hunk 2]
```diff
@@ -211,6 +225,25 @@ void CacheFuzzer::initSourceDataFiles() {
...
+        return;
+      }
+      std::random_device rd;
+      boost::random::uniform_int_distribution<int> dist(1, 100);
+      if (op->type == FaultFileOperation::Type::kWrite &&
+          dist(rd) <= kFileWriteErrorRate) {
+        VELOX_FAIL("Inject hook write failure");
+      }
+      if (op->type == FaultFileOperation::Type::kReadv &&
```
Comment (xiaoxmeng) [inline velox/exec/fuzzer/CacheFuzzer.cpp:241]: ssd cache all use readv? Thanks!

File: velox/exec/fuzzer/CacheFuzzer.cpp [hunk 3]
```diff
@@ -90,6 +99,9 @@ class CacheFuzzer {
 
  private:
   static constexpr int32_t kRandomized = -1;
+  static constexpr int32_t kFileWriteErrorRate = 1;
```
Comment (xiaoxmeng) [inline velox/exec/fuzzer/CacheFuzzer.cpp:102]: Both read/write set to 5%? Thanks!

File: velox/exec/fuzzer/CacheFuzzer.cpp [hunk 4]
```diff
@@ -418,6 +451,7 @@ void CacheFuzzer::resetSourceDataFiles() {
   if (fs_->exists(sourceDataDirPath)) {
     fs_->rmdir(sourceDataDirPath);
   }
+  faultyFileSystem()->clearFileFaultInjections();
```
Comment (xiaoxmeng) [inline velox/exec/fuzzer/CacheFuzzer.cpp:454]: Why clear injection here?

**Review by xiaoxmeng — COMMENTED** [review]
@zacw7 LGTM % minors. Thanks!

File: velox/exec/fuzzer/CacheFuzzer.cpp (hunk 1 above)
Comment (zacw7) [reply velox/exec/fuzzer/CacheFuzzer.cpp:232]: The input files are in the same directory, named as `file_{}` while cache files are named as `cache_{}`.

File: velox/exec/fuzzer/CacheFuzzer.cpp (hunk 2 above)
Comment (zacw7) [reply velox/exec/fuzzer/CacheFuzzer.cpp:241]: actually evictlog uses read. let me add that case to the injection.

File: velox/exec/fuzzer/CacheFuzzer.cpp [hunk 5]
```diff
@@ -90,6 +98,7 @@ class CacheFuzzer {
 
  private:
   static constexpr int32_t kRandomized = -1;
+  static constexpr int32_t kFileFaultInjectionRate = 5;
```
Comment (xiaoxmeng) [inline velox/exec/fuzzer/CacheFuzzer.cpp:101]:
```
s/kFileFaultInjectionRate/kFileFaultInjectionPct/
```

**Review by xiaoxmeng — APPROVED** [review]
@zacw7 LGTM. Thanks!

**Pull request closed without merging**